core/peersholder/peersHolder.go

```diff
@@ -80,13 +85,16 @@ func (ph *peersHolder) Remove(peerID core.PeerID) {
 	ph.Lock()
 	defer ph.Unlock()
 
-	delete(ph.peerIDs, peerID)
-
-	shard, index, found := ph.getShardAndIndexForPeer(peerID)
-	if found {
-		ph.removePeerFromMapAtIndex(shard, index)
+	_, found := ph.peerIDs[peerID]
+	if !found {
+		return
 	}
 
+	shard, index, _ := ph.getShardAndIndexForPeer(peerID)
+	ph.removePeerFromMapAtIndex(shard, index)
+
+	delete(ph.peerIDs, peerID)
+
 	pubKeyForPeerID := ""
 	for pubKey, peerInfo := range ph.pubKeysToPeerIDs {
```

Comment:
this for can be eliminated, at the expense of memory used by saving the public key in the peerIDPosition DTO (and renamed the DTO). What do you think, @AdoAdoAdo ?

Reply:
done